R/app_server.R

```diff
@@ -5,39 +5,37 @@
 #' @import shiny
 #' @noRd
 app_server <- function(input, output, session) {
-  # Initialize the database connection
+  
+  # Get R_CONFIG_ACTIVE from session data of Connect environment
+  # if it hasn't been set before calling run_app()
+  cat("Session group on Connect:", session$groups, " \n")
+  if (Sys.getenv("R_CONFIG_ACTIVE") == "rsconnect") {
+    Sys.setenv("R_CONFIG_ACTIVE" = set_trust_config(session$groups))
+  }
+  cat("Trust name:", get_golem_config("trust_name"), " \n")
 
-  pool <- odbc::dbConnect(
-    drv = odbc::odbc(),
-    driver = Sys.getenv("odbc_driver"),
-    server = Sys.getenv("HOST_NAME"),
-    UID = Sys.getenv("DB_USER"),
-    PWD = Sys.getenv("MYSQL_PASSWORD"),
-    database = "TEXT_MINING",
-    Port = 3306
-  )
+  # Initialize the database connection
+  pool <- get_pool()
 
   # fetch  the data
-
-  db_data <- dplyr::tbl(
-    pool,
-    dbplyr::in_schema(
-      "TEXT_MINING",
-      get_golem_config("trust_name")
-    )
-  ) %>%
-    tidy_all_trusts()
+  db_data <- get_db_data(pool, get_golem_config("trust_name"))
 
   # find out if there is data in the table
-
   data_exists <- db_data %>%
     dplyr::tally() %>%
     dplyr::pull(n) > 0
-  
-  if (!data_exists){
+
+  if (!data_exists) {
     showModal(modalDialog(
-      title = "NO DATA!",
-      HTML("<strong>Please go to the 'Data Management' tab to upload data.</strong>"),
+      title = strong("WELCOME!!!"),
```

Comment:
Can we change this to "Welcome!" instead?

Reply:
should we add the Trust name here e.g
Welcome to Nottinghamshire NHS Trust, Patient Experience Qualitative Data Categorisation Dashboard

Reply:
I'd avoid. Sometimes orgs change their name and it could get fiddly